docs/zkapps/advanced-snarkyjs/permissions.mdx

```diff
@@ -202,13 +190,13 @@ Now that you have provided a valid signature and the smart contract requires a v
 ### Upgradeability of smart contracts
 
 Another important part of smart contract development is upgradeability.
-By using permissions, we can either make our smart contract not upgradeable at all or upgradeable! 
-On Mina, when we deploy a smart contract we generate a verification key from our contract source code, which are then be stored on-chain and used to verify proofs that belong to that smart contract.
+By using permissions, you can make our smart contract not upgradeable at all or upgradeable! 
+On Mina, when you deploy a smart contract you generate a verification key from the contract source code. The verification key and the smart contract are stored on-chain and used to verify proofs that belong to that smart contract.
 Remember the permission called `setVerificationKey`? You can modify the authorization for this permission to set the upgradability of the smart contract. 
 
-#### Impossible
+#### Impossible to upgrade example
```

Comment:
A title structure like `Example: Impossible to upgrade` and `Example: Upgradeable with a proof` could be considered to help avoid any readers reading it in an unintended way. We've used a similar pattern for time locked account examples, since there were a few examples together and it helped group them.

Reply:
But I also am reviewing a diff, so maybe that doesn't work for readability in this case if it needs to be section titles. Just a suggestion to consider if it works